### ISSUE TITLE
LabelSync: Needs some performance tuning

### ISSUE THREAD

**Issue by cculianu** [Electron-Cash/Electron-Cash#1171]
In particular when using LabelSync with a high frequency of tx's (as happens in CashShuffle) -- it tends to do a lot of redundant network calls for each tx being sent out.

This normally wouldn't be a problem but when using an automated scheme to send tx's (such as CashShuffle) -- the speed at which tx's get generated overwhelms LabelSync.

LabelSync needs to be refactored to batch updates so as to avoid the overhead of a round-trip http-post per TX sent.

Also, I believe when loading/importing wallet labels from disk, LabelSync ends up doing a round-trip HTTP post *per label* updated.

So, in short: **LabelSync needs refactoring to be faster**.
